Prevent Sentry from blocking the F-Droid build

Prevent Sentry from blocking the F-Droid build. Either by disabling it for F-Droid or making it work. We are 1 year (one year) late with the F-Droid version
https://gitlab.com/pierre.slamich/fdroiddata/-/jobs/1028085676#L2588

We're thinking of creating a dimension `fdroid` to remove Sentry for F-Droid and deblock the year old failing build for Open Food Facts. Given that we already have 4 flavors, I'm a bit reluctant.

The DSN is already present in the manifest: https://github.com/openfoodfacts/openfoodfacts-androidapp/blob/b9db22ebafc424312b4ae4684b948c9db74ce7f7/app/src/main/AndroidManifest.xml#L44

But it's not present in sentry.properties
https://github.com/openfoodfacts/openfoodfacts-androidapp/blob/develop/sentry.properties

The error happens on mapping uploads during the F-Droid builds:
```
> compressing mappings
> uploading mappings
error: API request failed
  caused by: sentry reported an error: Authentication credentials were not provided. (http status: 401)
Add --log-level=[info|debug] or export SENTRY_LOG_LEVEL=[info|debug] to see more output.
Please attach the full debug log to all bug reports.
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:persistSentryProguardUuidsForOffReleaseOff-release'.
> Process 'command '/tmp/.sentry-cli8515915559930329791.exe'' finished with non-zero exit value 1
* Try:
``` 

Having seen https://github.com/dmitriy-ilchenko/QrAndBarcodeScanner/issues/49 and this commit (https://gitlab.com/fdroid/fdroiddata/-/merge_requests/7674/diffs?commit_id=1abd81c8589ee13898f1bdac29eb9857f923f19d), I felt hopeful we could pass the key in the build config on F-Droid. Is that possible ?

My understanding is that uploading needs AUTH while the APK only needs the URL where to upload crashes. 

- [ ] If so, can we disable uploading ? 
- [ ] What's the chance we won't be able to deobfuscate traces ?
- [ ] Is there a way to do a proper upload of symbol mapping within F-Droid ?


For reference, Sentry doc on the various ways to authenticate:
https://docs.sentry.io/platforms/java/configuration/

--- a/app/build.gradle.kts
+++ b/app/build.gradle.kts
@@ -13,20 +13,10 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-buildscript {
-    repositories {
-        mavenCentral()
-        jcenter()
-        google()
-        maven("https://jitpack.io")
-    }
-}
 plugins {
     id("com.android.application")
     id("de.timfreiheit.resourceplaceholders.plugin")
     id("org.greenrobot.greendao")
-    id("io.sentry.android.gradle")
     id("kotlin-android")
     id("kotlin-parcelize")
     id("kotlin-kapt")
@@ -207,22 +197,18 @@ dependencies {
     androidTestImplementation("tools.fastlane:screengrab:1.2.0")
 
     resourcePlaceholders { files = listOf("xml/shortcuts.xml") }
-
 }
-
 
 android {
     compileSdkVersion(30)
 
     testBuildType = obtainTestBuildType()
 
-
     buildFeatures {
         dataBinding = true
     }
 
     flavorDimensions("versionCode", "platform")
-
 
     defaultConfig {
         applicationId = "openfoodfacts.github.scrachx.openfood"
@@ -238,7 +224,6 @@ android {
         ndk.abiFilters("armeabi-v7a", "arm64-v8a", "x86", "x86_64")
 
         multiDexEnabled = true
-        // jackOptions.enabled = true
     }
 
     signingConfigs {
@@ -343,7 +328,6 @@ android {
         }
     }
 
-
     dexOptions {
         preDexLibraries = false
         javaMaxHeapSize = "4g"
@@ -381,7 +365,6 @@ android {
         // avoid "Method ... not mocked."
         unitTests.isReturnDefaultValues = true
         execution = "ANDROIDX_TEST_ORCHESTRATOR"
-
     }
 }
 
@@ -392,9 +375,3 @@ kapt {
 }
 
 greendao { schemaVersion(22) }
-
-
-
-
-
-

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -37,8 +37,6 @@ buildscript {
 
         classpath("com.google.dagger:hilt-android-gradle-plugin:$hiltVersion")
 
-
-        classpath("io.sentry:sentry-android-gradle-plugin:1.7.36")
         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
     }
 }

--- a/fastlane/Fastfile
+++ b/fastlane/Fastfile
@@ -69,20 +69,19 @@ lane :release do
         skip_upload_screenshots: true
     )
 
+    FLAVOR = SharedValues::GRADLE_FLAVOR + SharedValues::GRADLE_BUILD_TYPE
+    sentry_upload_proguard(
+      org_slug: 'openfoodfacts',
+      project_slug: 'openfoodfacts-android',
+      android_manifest_path: './app/build/intermediates/merged_manifests/' + FLAVOR + '/out/AndroidManifest.xml'
+      mapping_path: './app/build/outputs/mapping/' + FLAVOR + '/mapping.txt',
+    )
+
     sentry_create_release(
         org_slug: 'openfoodfacts',
         project_slug: 'openfoodfacts-android',
         version: PACKAGE_NAME + '@' + versionName + '+' + versionCode,
         finalize: false # Release will be finalized by the "finalize_sentry" lane, when the version is in production
-    )
-
-    FLAVOR = SharedValues::GRADLE_FLAVOR + SharedValues::GRADLE_BUILD_TYPE
-    sentry_upload_proguard(
-      auth_token: ENV['SENTRY_AUTH_TOKEN'],
-      org_slug: 'openfoodfacts',
-      project_slug: 'openfoodfacts-android',
-      android_manifest_path: './app/build/intermediates/merged_manifests/' + FLAVOR + '/out/AndroidManifest.xml'
-      mapping_path: './app/build/outputs/mapping/' + FLAVOR + '/mapping.txt',
     )
 
     sentry_set_commits(

--- a/fastlane/Fastfile
+++ b/fastlane/Fastfile
@@ -69,12 +69,20 @@ lane :release do
         skip_upload_screenshots: true
     )
 
-
     sentry_create_release(
         org_slug: 'openfoodfacts',
         project_slug: 'openfoodfacts-android',
         version: PACKAGE_NAME + '@' + versionName + '+' + versionCode,
         finalize: false # Release will be finalized by the "finalize_sentry" lane, when the version is in production
+    )
+
+    FLAVOR = SharedValues::GRADLE_FLAVOR + SharedValues::GRADLE_BUILD_TYPE
+    sentry_upload_proguard(
+      auth_token: ENV['SENTRY_AUTH_TOKEN'],
+      org_slug: 'openfoodfacts',
+      project_slug: 'openfoodfacts-android',
+      android_manifest_path: './app/build/intermediates/merged_manifests/' + FLAVOR + '/out/AndroidManifest.xml'
+      mapping_path: './app/build/outputs/mapping/' + FLAVOR + '/mapping.txt',
     )
 
     sentry_set_commits(

